ultravox/training/train.py

```diff
@@ -310,17 +311,27 @@ def train(args: config_base.TrainConfig):
         logging.info("Starting training...")
         t_start = datetime.now()
         logging.info(f"train start time: {t_start}")
+
         if args.val_steps:
-            trainer.evaluate()
+            if args.use_fsdp:
```

Comment:
Can you explain what's happening here? Why isn't the model initialized with FSDP?

Reply:
In the above, we removed `.to(device)` in favour of transformers/accelerate doing that for us when wrapping the `model` object with `FSDP`.

That means the model is now on CPU which is not great if you want to run validation. It will be moved to GPU only after the `trainer.train` call.

We could wrap the model ourselves too if need be explicitly. Here's the code (we might be able to borrow `FullyShardedDataParallelPlugin` object to reuse the `set_auto_wrap_policy` function), but for now I had decided not to do that. Not a strong opinion either way.

<img width="598" alt="image" src="https://github.com/user-attachments/assets/b91c74b1-56ee-4093-aed4-bd7b0225d88d">
<img width="858" alt="image" src="https://github.com/user-attachments/assets/b0a67155-d282-46ab-8cb1-c407369175ae">

Reply:
So... does this mean we can't run `evaluate` for fsdp models, unless we add manual wrapping? Were you planning to revisit this in a followup?

Reply:
`evaluate` (as in an evaluation that runs automatically after training) doesn't work either way now and I didn't want to spend time on it until I knew what ZQ wanted to do.

To clarify, the numbers I provided were using vllm as the inference engine.